src/main/java/com/softserve/demo/service/impl/CustomerServiceImpl.java

```diff
@@ -72,7 +81,51 @@ public void addImageToCustomer(Integer id, String fileName) {
     }
 
     @Override
-    public CustomerDTO findCustomerByUserId(Integer id) {
-        return customerMapper.customerToCustomerDTO(customerRepository.findCustomerByUserId(id));
+    public CustomerDTO getCustomerByUserId(Integer id) {
+        return customerMapper.customerToCustomerDTO(customerRepository.getCustomerByUserId(id));
+    }
+
+    @Override
+    public Page<CustomerDTO> getCustomersByStatus(Pageable pageable, CustomerStatus status) {
+        return customerRepository.findByStatus(status,pageable).map(customerMapper::customerToCustomerDTO);
+    }
+
+    @Override
+    public CustomerDTO updateStatus(Integer id, String status) {
+        Customer customer = customerRepository.findById(id).orElseThrow(() -> new NotFoundException("Customer not found"));
+        customer.setStatus(CustomerStatus.valueOf(status));
+        customerRepository.save(customer);
+        return customerMapper.customerToCustomerDTO(customer);
+    }
+
+    @Override
+    public <T> Page<Customer> findAll(Specification<T> approved, int page, int numberOfProvidersOnPage, String sortBy) {
+        return customerRepository.findAll(approved,PageRequest.of(page, numberOfProvidersOnPage, Sort.by(sortBy).descending()));
+    }
+
+    @Override
+    public void addFavourite(Integer id, ProviderDTO providerDTO) {
+        Customer customer = customerRepository.findCustomerById(id);
+        List<Provider> list = customer.getFavourite();
+        list.add(providerMapper.providerDTOToProvider(providerDTO));
+        customer.setFavourite(list);
+        customerRepository.save(customer);
+    }
+
+    @Override
+    public void removeById(Integer customerId, Integer favouriteId) {
+        Customer customer = customerRepository.findCustomerById(customerId);
+        List<Provider> list = customer.getFavourite();
+        int count = 0;
+        for (Provider provider : list) {
+            count++;
+            if (provider.getId().equals(favouriteId)) {
+                break;
+            }
+        }
+
```

Comment:
Why do you need this count here?

Reply:
using this "count" helps to know which element of list i need to remowe

Reply:
You can delete it inside the cycle

Reply:
i understand that it looks strang but as i remember there vere some mistakes because of which i wasn`t able to use streams